### PR TITLE
{*}[intel/2015.08] HPL 2.1 (REVIEW)

### PR DESCRIPTION
**Pull request by boegel**
latest version of the 2015 Intel compilers + latest Intel MPI and MKL

requires ~~#2108~~ #2214

### CODE REVIEW

**Comment by boegel** [comment]
@wpoely86: please review?

this is a good candidate for `intel/2016a` imho

File: easybuild/easyconfigs/i/intel/intel-2015.08.eb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+homepage = 'http://software.intel.com/en-us/intel-cluster-toolkit-compiler/'
+description = """Intel Cluster Toolkit Compiler Edition provides Intel C/C++ and Fortran compilers, Intel MPI & Intel MKL."""
+
+toolchain = {'name': 'dummy', 'version': 'dummy'}
+
+compver = '2015.5.223'
+gccsuff = '-GCC-4.9.3-2.25'
+
+dependencies = [
+    ('icc', compver, gccsuff),
```
Comment (wpoely86) [inline easybuild/easyconfigs/i/intel/intel-2015.08.eb]: Isn't it clear to also add `GCC`/`GCC-Core` here?

Comment (boegel) [inline easybuild/easyconfigs/i/intel/intel-2015.08.eb]: good suggestion, it'll make the dependency more explicit; same for `binutils`?

Comment (wpoely86) [inline easybuild/easyconfigs/i/intel/intel-2015.08.eb]: yes

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 8 out of 8 (8 easyconfigs in this PR)
Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/9073a794441b35bd3642 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 8 out of 8 (8 easyconfigs in this PR)
Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/937c212884e9211b99b7 for a full test report.

File: easybuild/easyconfigs/i/icc/icc-2015.5.223-GCC-4.9.3-2.25.eb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+version = '2015.5.223'
+
+homepage = 'http://software.intel.com/en-us/intel-compilers/'
+description = "C and C++ compiler from Intel"
+
+toolchain = {'name': 'dummy', 'version': 'dummy'}
+
+sources = ['l_ccompxe_%(version)s.tgz']
+
+gcc = 'GCC'
```
Comment (boegel) [inline easybuild/easyconfigs/i/icc/icc-2015.5.223-GCC-4.9.3-2.25.eb]: as pointed out by @ocaisa in #2214, this should be `GCCcore`, and `binutils` should be included as well as a dep

File: easybuild/easyconfigs/i/ifort/ifort-2015.5.223-GCC-4.9.3-2.25.eb [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+version = '2015.5.223'
+
+homepage = 'http://software.intel.com/en-us/intel-compilers/'
+description = "Fortran compiler from Intel"
+
+toolchain = {'name': 'dummy', 'version': 'dummy'}
+
+sources = ['l_fcompxe_%(version)s.tgz']
+
+gcc = 'GCC'
```
Comment (boegel) [inline easybuild/easyconfigs/i/ifort/ifort-2015.5.223-GCC-4.9.3-2.25.eb]: as pointed out by @ocaisa in #2214, this should be `GCCcore`, and `binutils` should be included as well as a dep

File: easybuild/easyconfigs/i/intel/intel-2015.08.eb [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+toolchain = {'name': 'dummy', 'version': 'dummy'}
+
+compver = '2015.5.223'
+
+gccver = '4.9.3'
+binutilsver = '2.25'
+gccsuff = '-GCC-%s-%s' % (gccver, binutilsver)
+dependencies = [
+    ('GCCcore', gccver),
+    ('binutils', binutilsver, '-GCCcore-%s' % gccver),
```
Comment (boegel) [inline easybuild/easyconfigs/i/intel/intel-2015.08.eb]: @ocaisa: I got it partially right ;)

File: easybuild/easyconfigs/i/ifort/ifort-2015.5.223-GCC-4.9.3-2.25.eb [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+gccver = '4.9.3-2.25'
+versionsuffix = '-%s-%s' % (gcc, gccver)
```
Comment (boegel) [inline easybuild/easyconfigs/i/ifort/ifort-2015.5.223-GCC-4.9.3-2.25.eb]: it's probably cleaner to stick to using `GCC` as a label in the versionsuffix, but using `GCCcore` as a dep? @ocaisa?

**Comment by boegel** [comment]
Jenkins: test this please

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 8 out of 8 (8 easyconfigs in this PR)
Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/e8adadd529b5ffcdc9ad for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 8 out of 8 (8 easyconfigs in this PR)
Linux centos linux 7.1.1503, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/09ecd995606cfd76a58c for a full test report.

**Comment by boegel** [comment]
Good to go, thanks for the review @wpoely86!

**Pull request merged**